tests/fsharpqa/Source/Conformance/SpecialAttributesAndTypes/Imported/CallerInfo/CallerMemberName.fs

```diff
@@ -45,6 +47,8 @@ type MyTy() =
             |> List.head
         MyTy.Check(result, Some("CheckMembers"), "lambda")
         MyTy.Check(functionVal (), Some("functionVal"), "functionVal")
+        MyTy.Check(anonymousLambda, Some("anonymousLambda"), "anonymousLambda")
+        MyTy.Check(asyncVal, Some("asyncVal"), "asyncVal")
```

Comment:
I see.  There's logic to that for sure. 

What about 

* "do" or other top-level side effect within a module
* "static do" within a class type

I glanced again just now and didn't spot tests for those